test/client/connection.js

```diff
@@ -90,4 +91,71 @@ describe('client connection', function() {
     });
   });
 
+  describe('state management using setSocket', function() {
+    it('initial connection.state is connecting, if socket.readyState is CONNECTING', function () {
+        // https://html.spec.whatwg.org/multipage/web-sockets.html#dom-websocket-connecting
+        var socket = { readyState: 0 }
```

Comment:
Style-wise, we tend not to put spaces inside of brackets, and use semicolons. I'll just go ahead and clean up after merging.